system/include/simd128.h

```diff
@@ -0,0 +1,753 @@
+/*
+WebAssembly SIMD128 Intrinsics
+*/
+
+#include <stdint.h>
+
+// User-facing types
+typedef int8_t v128 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef int8_t i8x16 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef uint8_t u8x16 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef int16_t i16x8 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef uint16_t u16x8 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef int32_t i32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef uint32_t u32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef float f32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+
+#ifdef __wasm_unimplemented_simd128__
+
+typedef int64_t i64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef uint64_t u64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef double f64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+
+#endif // __wasm_unimplemented_simd128__
+
+// Internal types
+typedef int8_t __v128_u __attribute__((__vector_size__(16), __aligned__(1)));
+typedef char __v128 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef char  __i8x16 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef short __i16x8 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef int __i32x4 __attribute__((__vector_size__(16), __aligned__(16)));
+typedef long long __i64x2 __attribute__((__vector_size__(16), __aligned__(16)));
+
+#define __DEFAULT_FN_ATTRS __attribute__((__always_inline__, __nodebug__, __target__("simd128"), __min_vector_width__(128)))
+
+// v128 wasm_v128_load(void* mem)
+static __inline__ v128 __DEFAULT_FN_ATTRS wasm_v128_load(void* __mem) {
```

Comment:
should be `const void* __mem` to allow loads via const-pointers

Reply:
Done, thanks!